README.md

```diff
@@ -1179,6 +1187,32 @@ Prefer: return=minimal
 HTTP/1.1 204 No Content
 ```
 
+## HTTP Methods
+
+Following HTTP Methods are supported currently.
+
+### HEAD
+The same as [GET](#get), except that it doesn't return any body.
+
+### GET
+Used to query a resource with its sub/embedded resources.
+
+### POST
+Used to create new resource document, where new `ID` is generated from the server.
+
+### PUT
+Used to create/update single resource document given its `ID`.\
+Be aware when dealing with resource fields with `Default` set. Initial creation for such resources will set particular field to its default value if omitted, however on subsequent `PUT` calls this field will be deleted if omitted. If persistent `Default` field is needed use `{Required: true}` with it. 
+
+### PATCH
+Used to update/patch single resource document given its `ID`.
+
```

Comment:
Should we include the move the note on JSON-Patch here?

Reply:
Yes, I guess we can. Will do it in a while.